utils/clickhouse.cc

```diff
@@ -0,0 +1,74 @@
+/**
+ *		Tempesta FW
+ *
+ * Copyright (C) 2024 Tempesta Technologies, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License,
+ * or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59
+ * Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+ */
+
+#include <iostream>
+
+#include <clickhouse/base/socket.h>
+#include <clickhouse/client.h>
+
+#include "clickhouse.h"
+
+
+constexpr size_t MAX_MSEC = 100;
+constexpr size_t MAX_EVENTS = 1000;
+
+static auto
+now_ms()
+{
+	return std::chrono::duration_cast<std::chrono::milliseconds>(
+			std::chrono::system_clock::now().time_since_epoch())
+		.count();
+}
+
+TfwClickhouse::TfwClickhouse(std::string host, std::string table_name,
+			     clickhouse::Block *(*cb)())
+{
+	block_callback_ = cb;
+	table_name_ = table_name;
+
+	client_ = std::make_unique<clickhouse::Client>(
+		clickhouse::ClientOptions().SetHost(std::move(host)));
+	block_ = cb();
```

Comment:
It seems that we should have destructor to delete block_

Reply:
> Also it seems that we can return std::uniqui_ptr from callback. to not use delete.
Sorry, I met C++ before uniqui_ptr was presented.
Is it fast enough for 'hot' places?

Reply:
Yeah, a good concern. Could you please make a short example at https://godbolt.org/ and observe how modern clang and gcc compile unique_ptr? I reckon it should be free, but some wrapping stuff still could be generated by the compilers...

Reply:
We need to share pointer to the code outside TfwClickhouse class so we need to use shared_ptr. 
Every access costs us a call:
`call    std::__shared_ptr_access<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, (__gnu_cxx::_Lock_policy)2, false, false>::operator*() const`